dataline-integration-tests/src/test/java/io/dataline/integration_tests/destinations/BaseIntegrationTestCase.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Dataline
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.dataline.integration_tests.destinations;
+
+import io.dataline.workers.process.DockerProcessBuilderFactory;
+import io.dataline.workers.process.ProcessBuilderFactory;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+
+public abstract class BaseIntegrationTestCase {
+
+  private static final Path TESTS_PATH = Path.of("/tmp/dataline_integration_tests");
+
+  protected Path jobRoot;
+  protected Path workspaceRoot;
+  protected ProcessBuilderFactory pbf;
+
+  @BeforeEach
+  public void setUp() throws IOException {
+    Files.createDirectories(TESTS_PATH);
+    workspaceRoot = Files.createTempDirectory(TESTS_PATH, "dataline-integration");
+    jobRoot = Path.of(workspaceRoot.toString(), "job");
+    Files.createDirectories(jobRoot);
+
+    pbf = new DockerProcessBuilderFactory(workspaceRoot, workspaceRoot.toString(), "host");
+  }
+
+  @AfterEach
+  public void tearDown() throws IOException {
+    FileUtils.deleteDirectory(new File(workspaceRoot.toUri()));
```

Comment:
it may be useful to leave the workspace since it's already a temp directory that the OS sweeps in case we need a deeper look at logs

Reply:
We can disable that on a case by case basis if we need a deeper look.